### DISCUSSION TITLE
ANSYS: Access denied error

### DISCUSSION THREAD

**Discussion by rjwilliams84** [ansys/pyaedt#2231] [Q&A] [answered]
I just recently found pyaedt and wanted to give it a try with my install of Ansys Electronics Desktop 2022 R2.

I managed to install pyaedt and started writing a test script in spyder to get an understanding of how to use the different functions in the pyaedt package. The script I wrote consists of just the import command and the Desktop() command to open up ANSYS, however when I run this script (either through spyder or command line) ANSYS starts to open but then immediately throws an error stating that access to the folder that my script is in is denied. 

I am the only user on my machine (windows workstation) and am the administrator. I'm not sure what to do/where to investigate this issue, so thought I'd post it here to see if anyone has any ideas or has encountered this before. I could not find anything regarding this on the pyaedt website, nor the ANSYS website.

Thanks!

**Comment by Devin-Crawford** [comment]
Have you tried running one of the examples?  Also, it's best to use a Python interpreter with user access.  If you're using the system Python, you might try making a [virtual environment](https://docs.python.org/3/library/venv.html) and install dependencies there.

**Reply by rjwilliams84** [reply]
I have tried the HFSS_Dipole.py example. First thing with that example is that I get the error that module 'pyaedt' has no attribute 'launch_desktop' at line 31. 

If I change line 31 to d = pyaedt.Desktop("2022.2", non_graphical=non_graphical, new_desktop_session=True), the script starts running but when it launches AEDT I get the same error: "Access to C:/Users/[username]/Documents/Python Scripts/ was denied"

The above error is presented as given on my machine, aside from [username]. To be clear, this error is from Ansys Electronics Desktop, not python, and the folder to which it says access was denied is the folder I am running the HFSS_Dipole.py example from.

In my Spyder (IDE I'm using) console I see the following when I run the script, prior to AEDT launching (and throwing the access denied error):

pyaedt INFO: using existing logger.
pyaedt INFO: Launching PyAEDT outside AEDT with CPython and PythonNET.
pyaedt INFO: AEDT installation Path C:\Program Files\AnsysEM\v222\Win64.
pyaedt INFO: Launching AEDT with module PythonNET.

**Comment by maxcapodi78** [comment]
@rjwilliams84 the missing launch_desktop is probably due to the fact that you are using an older version of pyaedt. This method has been added recently in pyaedt so it would be nice to run

```
pip install pyaedt -U 
```

then the second error is most probably due to the fact that you run the script from Spyder folder and this folder is owned by spyder only with no access to other tools/ script.
maybe you can change dir to temp at beginning of your script

```
import os
os.chdir(r'"C:\temp")

import pyaedt
...
```

**Reply by rjwilliams84** [reply]
This has been somewhat helpful. I did have an older version of pyaedt and was able to update to the newest version using pip.

I added the line os.chdir(r"C:\temp") as suggested. This did alleviate the "access denied" error, so you must have been correct that spyder was not allowing access to other tools.

However, I now have a new error, how fun. Although I suspect you will have an idea on what the solution is. Now that I change the directory to C:\temp, I can launch AEDT but now get a message dialog saying "File is empty: C:/temp/" and then when I click the "OK" button AEDT automatically closes.

So some progress, but still some bumps in the road. 

And in case it was not clear, I was running the HFSS_dipole.py example.

As another test, I moved the HFSS_dipole.py script to the C:\temp\folder to see if the script not being in there was an issue, however when I did that AEDT began throwing the "Access Denied" error again, but for C:\temp.

**Comment by rjwilliams84** [comment, marked as answer]
The issue was the way pyaedt was communicating with my install of hfss. Using grpc was the solution that worked for me, particularly adding in this line of code in the beginning of my python script:

pyaedt.settings.use_grpc_api=True

This allowed me to successfully run the HFSS_Dipole.py example. Thank you to Massimo at ANSYS for helping me work through this issue. If you are experiencing a similar issue to me, please try adding in the above line of code.

**Reply by Devin-Crawford** [reply]
That's great to hear since grpc is the way of the future.  Glad it's working.
